src/Microsoft.Data/DataFrame.Arrow.cs

```diff
@@ -0,0 +1,165 @@
+                        dataFrameColumn = new PrimitiveDataFrameColumn<byte>(field.Name, byteValueBuffer, byteNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.UInt16:
+                        PrimitiveArray<ushort> arrowUshortArray = (PrimitiveArray<ushort>)arrowArray;
+                        ReadOnlyMemory<byte> ushortValueBuffer = arrowUshortArray.ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> ushortNullBitMapBuffer = arrowUshortArray.NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveDataFrameColumn<ushort>(field.Name, ushortValueBuffer, ushortNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.UInt32:
+                        PrimitiveArray<uint> arrowUintArray = (PrimitiveArray<uint>)arrowArray;
+                        ReadOnlyMemory<byte> uintValueBuffer = arrowUintArray.ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> uintNullBitMapBuffer = arrowUintArray.NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveDataFrameColumn<uint>(field.Name, uintValueBuffer, uintNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.UInt64:
+                        PrimitiveArray<ulong> arrowUlongArray = (PrimitiveArray<ulong>)arrowArray;
+                        ReadOnlyMemory<byte> ulongValueBuffer = arrowUlongArray.ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> ulongNullBitMapBuffer = arrowUlongArray.NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveDataFrameColumn<ulong>(field.Name, ulongValueBuffer, ulongNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Decimal:
+                    case ArrowTypeId.Binary:
+                    case ArrowTypeId.Date32:
+                    case ArrowTypeId.Date64:
+                    case ArrowTypeId.Dictionary:
+                    case ArrowTypeId.FixedSizedBinary:
+                    case ArrowTypeId.HalfFloat:
+                    case ArrowTypeId.Interval:
+                    case ArrowTypeId.List:
+                    case ArrowTypeId.Map:
+                    case ArrowTypeId.Null:
+                    case ArrowTypeId.Struct:
+                    case ArrowTypeId.Time32:
+                    case ArrowTypeId.Time64:
+                    default:
+                        throw new NotImplementedException(nameof(fieldType.Name));
+                }
+                ret.InsertColumn(ret.ColumnCount, dataFrameColumn);
+                fieldIndex++;
+            }
+            return ret;
+        }
+
+        public IEnumerable<RecordBatch> ToArrowRecordBatches()
```

Comment:
Not sure if I have a strong reason to make this a static method or an extension

Reply:
I think it is fine like this.